src/surforama/utils/twoD_averages.py

```diff
@@ -0,0 +1,342 @@
+    )
+
+    # Flatten and rotate the grid, then add the point
+    local_grid_flat = local_grid.reshape(3, -1)
+    global_grid_flat = np.dot(rotation_matrix.T, local_grid_flat).T + point
+
+    # Use scipy's map_coordinates to extract the aligned volume
+    extracted_volume = map_coordinates(
+        tomogram, global_grid_flat.T, order=1, mode="nearest"
+    ).reshape(shape)
+
+    return extracted_volume
+
+
+def create_2D_averages(
+    positions: list,
+    mesh: trimesh.Trimesh,
+    tomogram: np.ndarray,
+    shape: tuple = (20, 20, 20),
+    mirror: bool = True,
+) -> np.ndarray:
+    """
+    Create 2D averages from a list of positions using a given mesh and tomogram.
+
+    Parameters
+    ----------
+    positions : list
+        A list of 3D points.
+    mesh :
+        A mesh object used to find normals corresponding to the given positions.
+    tomogram : np.ndarray
+        A 3D numpy array containing the tomogram data.
+    mirror : bool, optional
+        Flag to mirror the results horizontally for each 2D average.
+
+    Returns
+    -------
+    np.ndarray
+        An array of 2D averages calculated for each position.
+
+    """
+    averages = []
+    for position in positions:
+        normal = find_closest_normal(mesh, position)
+        volume = extract_normal_volume(position, normal, tomogram, shape)
+        avg = avg_vol_2D(volume, mirror=mirror)
+        averages.append(avg)
+    return np.array(averages)
+
+
+def normalize_averages(avgs: np.ndarray) -> np.ndarray:
+    """
+    Normalize the average images.
+
+    This is done by subtracting the mean and dividing by the standard deviation.
+    Mean and standard deviation are calculated across all the averages.
+
+    Parameters
+    ----------
+    avgs : np.ndarray
+        The array of averages to be normalized.
+
+    Returns
+    -------
+    np.ndarray
+        The normalized averages.
+    """
+    mean = np.mean(avgs)
+    std = np.std(avgs)
+    avgs = (avgs - mean) / std
+    return avgs
+
+
+def noshift_crosscorrelation(img1, img2, mask):
+    """This is also the cross-correlation used in PySeg:
+    https://github.com/anmartinezs/pyseg_system/blob/925f04b3415bff45c4bdd957d17e9a8cabc2fb8f/code/pyseg/sub/klass.py#L111
+    """
+    corr = np.sum(img1 * img2) / np.sum(mask)
+    return corr
+
+
+def extract_masked_image(image: np.ndarray, mask: np.ndarray) -> np.ndarray:
+    """
+    Extract a sub-image from the given volume using a mask.
+
+    Parameters
+    ----------
+    volume : np.ndarray
+        The full volume from which to extract.
+    mask : np.ndarray
+        The mask indicating the sub-volume to extract.
+
+    Returns
+    -------
+    np.ndarray
+        The extracted sub-volume defined by the mask.
+    """
+    rows, cols = np.where(mask == 1)
+    return image[rows.min() : rows.max() + 1, cols.min() : cols.max() + 1]
+
+
+def build_cc_matrix(avgs: np.ndarray, masks: list) -> np.ndarray:
```

Comment:
I guess all the cross-correlation and clustering stuff from the previous iteration is redundant now -- so best to remove it?